test/test_core.py

```diff
@@ -9229,7 +9229,7 @@ def test_pthread_offset_converter_modularize(self):
     self.set_setting('USE_OFFSET_CONVERTER')
     self.set_setting('MODULARIZE')
     self.set_setting('EXPORT_NAME', 'foo')
-    create_file('post.js', 'foo();')
+    create_file('post.js', 'if (typeof importScripts != "function") foo();')
```

Comment:
Will users need to add such changes too?

I was hoping not since the worker.js file loaded the main file including post-js stuff already, so I'm not sure why this is needed.

Reply:
I think this code is already basically broken since `--pre-js`/`--post-js` files are run on all threads.

Doing an unguarded `foo()` in your `extern-pre.js` is basically constructing a new instance when each worker is created (i.e. when the top level JS runs).

So this code as written seems wrong since manually creating an instance of `foo()` should only ever be done on the main thread. 

Do you think it worth devising a way to ignore this on pthread in case folks are doing this pattern?

Reply:
Makes sense this was already broken. But then how did it work before? It seems like the same problem should have happened.

Reply:
I think what was happening is that with the old code a new instance of the `foo()` module was being created (and then immediately discarded) when each worker started.   Anther instance (the one was then being created and used to the pthread entry point).

I will confirm this .. and perhaps looks for a workaround (perhaps user calls to the module constructor function should assert)

Reply:
What do you think about have `extern-pre` and `extern-post` code only run on the main thread?

Reply:
Landed in #21750 and reverted this.